### PR TITLE
lightning qubit uses parameter shift if metric tensor applied

### PR DESCRIPTION
**Pull request by albi3ro**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/workflow/qnode.py [hunk 1]
```diff
@@ -592,8 +592,17 @@ def _update_gradient_fn(self, shots=None, tape=None):
             return
         if tape is None and shots:
             tape = qml.tape.QuantumScript([], [], shots=shots)
+
+        diff_method = self.diff_method
+        if (
+            self.device.name == "lightning.qubit"
```
Comment (josh146) [inline pennylane/workflow/qnode.py:598]: quick question but do we need to validate against `lightning.kokkos` and `lightning.gpu` as well? Or do they not yet default to adjoint?

Comment (albi3ro) [reply pennylane/workflow/qnode.py:598]: They do not yet default to adjoint.

File: tests/gradients/core/test_metric_tensor.py [hunk 2]
```diff
@@ -1140,19 +1142,21 @@ def circuit(*params):
         if isinstance(mt, tuple):
             assert all(qml.math.allclose(_mt, _exp) for _mt, _exp in zip(mt, expected))
         else:
+            print(mt - expected)
```
Comment (dwierichs) [inline tests/gradients/core/test_metric_tensor.py:1145]:
```suggestion
```

File: pennylane/gradients/metric_tensor.py [hunk 3]
```diff
@@ -610,23 +605,16 @@ def _get_first_term_tapes(layer_i, layer_j, allow_nonunitary, aux_wire):
...
+            ops = [
+                qml.Hadamard(wires=aux_wire),
+                *layer_i.pre_ops,
+                gen_op_i,
+                *ops_between_cgens,
+                gen_op_j,
+            ]
+            new_tape = qml.tape.QuantumScript(ops, [qml.expval(qml.X(aux_wire))], shots=shots)
+
+            tapes.append(new_tape)
```
Comment (dwierichs) [inline pennylane/gradients/metric_tensor.py:617]: :rocket:

File: doc/releases/changelog-0.36.0.md [hunk 4]
```diff
@@ -551,6 +551,9 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* Patches the QNode so that parameter-shift will be considered best with lightning if
+  `qml.metric_tensor` is in the transform program.
```
[the remark this answers is not in the document]
Comment (dwierichs) [reply doc/releases/changelog-0.36.0.md:555]: ^outdated?

**Comment by albi3ro** [comment]
See issue #5642

**Review by mudit2812 — APPROVED** [review]

**Review by lillian542 — APPROVED** [review]

**Comment by trbromley** [comment]
Thanks @albi3ro! So I understand, this PR does not resolve the issue reported in #5642 but is just related, right?

**Comment by albi3ro** [comment]
When writing tests for this PR, I found #5642.  Because I didn't have time to fix it, I marked some jax tests as `xfail` and opened a bug report. 

So this PR uncovered but did not fix #5642 .

**Pull request merged**